language/move-core/types/src/value.rs

```diff
@@ -69,6 +69,12 @@ impl MoveFieldLayout {
 pub enum MoveStructLayout {
     /// The representation used by the MoveVM
     Runtime(Vec<MoveTypeLayout>),
+    /// The representation used by the MoveVM with extra unique identifier for distinct type.
+    CheckedRuntime {
```

Comment:
This is rather ugly.... IMHO the `Runtime` variant should never be here in the first place as why would we need anything related to runtime in the `move-core-types` crate? I understand this is because we had multiple representations of the same stuff but this was quite ugly if we want to add more runtime specific thing here due to the dependency issue. cc: @vgao1996